services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Utils.kt

```diff
@@ -120,3 +121,32 @@ fun String?.isValidBusinessId(): Boolean {
         false
     }
 }
+
+/**
+ * Valid OVT ("verkkolaskuosoitteen vastaanottajan identifioiva osa") requirements:
+ * 1. format 0037NNNNNNNNXXXXX, where 0037 = "Suomen verohallinnon tunnus", NNNNNNNN = business id
+ *    without the dash ('-') and XXXXX is an optional reference (0-5 characters).
+ *
+ * Only verifies that the OVT is of valid form. It does not guarantee that it actually exists. Does
+ * NOT verify the given business id, only that it is included in the OVT (or is null).
+ */
+fun String?.isValidOVT(businessId: String?): Boolean {
+    logger.info { "Verifying OVT: $this" }
+
+    if (this == null) {
+        logger.warn { "OVT is null." }
+        return false
+    }
+
+    val matchResult = ovtRegex.find(this)
+    if (matchResult == null) {
+        logger.warn { "Invalid format." }
+        return false
+    }
+    if (businessId != null && matchResult.groups[1]!!.value != businessId.replace("-", "")) {
```

Comment:
Is it guaranteed that the OVT has the business ID of the company? So it's not possible to use a billing processor, or if a billing processor is used, the OVT is still for the main company?

Reply:
Looks like ProCounter only accepts numbers as the OVT.

Reply:
Good point! I think I made unnecessary assumptions in my implementation. Maybe we need only to check that the length is correct (12-17 digits)?

Reply:
Based on the Slack conversation, the only check could be that it's at least 12 characters long.

Reply:
I changed the validation to only check the length of the string.